lib/membrane_framerate_converter.ex

```diff
@@ -73,9 +69,48 @@ defmodule Membrane.FramerateConverter do
 
   @impl true
   def handle_caps(:input, caps, _context, %{framerate: framerate} = state) do
+    state = %{state | input_framerate: caps.framerate}
+
     {{:ok, caps: {:output, %{caps | framerate: framerate}}}, state}
   end
 
+  @impl true
+  def handle_end_of_stream(:input, _ctx, %{input_framerate: {0, _denom}} = state) do
+    {{:ok, end_of_stream: :output}, state}
+  end
+
+  def handle_end_of_stream(:input, _ctx, %{last_buffer: nil} = state) do
+    {{:ok, end_of_stream: :output}, state}
+  end
+
+  @impl true
+  def handle_end_of_stream(
+        :input,
+        _ctx,
+        %{last_buffer: last_buffer} = state
+      ) do
+    input_frame_duration = get_frame_duration(state.input_framerate)
+    output_frame_duration = get_frame_duration(state.framerate)
+    # calculate last target timestamp so that the output video duration is closest to original
+    best_last_timestamp = last_buffer.pts + input_frame_duration - div(output_frame_duration, 2)
+    buffers = fill_to_last_timestamp(best_last_timestamp, state)
+    {{:ok, [buffer: {:output, buffers}, end_of_stream: :output]}, state}
+  end
+
+  defp get_frame_duration({num, denom}) do
+    round(Membrane.Time.second() * denom / num)
```

Comment:
To make the calculations consistent, I'd use Ratio here and use this function for both bumping timestamp and best_last_timestamp calculation